get_anndata.py

```diff
@@ -4,31 +4,26 @@
 This script will
 
 - open anndata expression files
-- open covariate files
+- open cell and samples covariate files
 - create pheno_cov_files
 - create cis window files
 
 these files will be used as inputs for the
-SAIGE-QTL association pipeline.
+SAIGE-QTL association pipelines (both common and rare).
 
 To run:
 
-In main:
-
-In test:
-
 analysis-runner \
-    --description "make expression input files" \
-    --dataset "bioheart" \
-    --access-level "test" \
-    --output-dir "saige-qtl/bioheart_n990_and_tob_n1055/input_files" \
-    --image australia-southeast1-docker.pkg.dev/cpg-common/images/scanpy:1.9.3 \
-    python3 get_anndata.py --celltypes B_naive --chromosomes chr21 \
-    --anndata-files-prefix gs://cpg-bioheart-test/saige-qtl/anndata_objects_from_HPC \
-    --celltype-covs-files-prefix gs://cpg-bioheart-test/saige-qtl/celltype_covs_from_HPC \
-    --sample-covs-file gs://cpg-bioheart-test-analysis/saige-qtl/input_files/covariates/sex_age_geno_pcs_shuffled_ids_tob_bioheart.csv \
-    --concurrent-job-cap=1000 --pc-job-cpu=1 --cis-job-cpu=1
-
+   --description "make expression input files" \
+   --dataset "bioheart" \
+   --access-level "full" \
```

Comment:
Do you need `full` access, or is `main` enough? (Or whatever the level is called in between test and full)

Reply:
That is one of the main questions I had! I used to always run everything with `standard`, get some permission error, try again with `full` and it would work! But I could never remember for which of the scripts that was, and as it turns out is every one of them (I think)?

It was definitely always true for the `get_sample_covariates.py` script (something to do with metamist) but now it's all scripts. Matt or others would know for sure but my guess of the culprit is [this type of statement ](https://github.com/populationgenomics/saige-tenk10k/blob/main/make_group_file.py#L163-L167) or in general lines checking for existence of the file before I run?

Reply:
For the record, I'll list below the failed logs when running with `standard`:

* `get_genotype_vcf.py`: https://batch.hail.populationgenomics.org.au/batches/483824/jobs/1
* `get_sample_covariates.py`: https://batch.hail.populationgenomics.org.au/batches/485730/jobs/1
* `get_anndata.py`: https://batch.hail.populationgenomics.org.au/batches/485749/jobs/1

For the first two, it may be because I am wanting to access `test` objects, for the get anndata script it's because I'm wanting to copy to `tmp`?

@MattWellie would you be able to comment?

Reply:
All pretty irritating issues - `test` only has access to `test`, `standard` has access to `main`, `full` has access to both. To resolve the need for `full` when running read-only processes that are looking at test data, you should probably move the data you're reading into `main` (this would also futureproof it by making it ~impossible to delete)

The `get_anndata.py` is really sucky - when you do a gsutil cp the client usually stages files (or file fragments locally) during the copy, then deletes the temp files. This isn't permitted with `standard` permissions, as that permission level can't delete any data from `main`, even from the `main-tmp` bucket. TBH this is a real user friction oversight, and we should probably resolve this by making `-tmp` buckets more like scratch space with lower/no write protection... In the short term, the solution is having to run with `full` permissions, or find an alternative form of the copy that doesn't attempt to clean up its own temp files :/

Reply:
After discussing this morning with @MattWellie and @Alexander-Stuckey we agreed that for now there is no obvious solution for the `get_anndata.py` and SAIGE-QTL runner scripts (`saige_assoc.py` and `saige_assoc_set_test.py`), as they require copying and deleting from the `main-tmp` bucket.

I have updated the `get_genotype_vcf.py` script to take a file in `main-analysis` in with info on related individuals and have updated the instructions to require `standard` permissions only.

The `get_sample_covariates.py` script needs a little more work as I need to take in new `main-analysis` Hail Table files instead of temporary CSV / TSV (one of which in `test`) so I have done that in a separate PR (https://github.com/populationgenomics/saige-tenk10k/pull/151).

`make_group_file.py` ran fine in `standard`: https://batch.hail.populationgenomics.org.au/batches/485958 so updating that as well